addons/website_livechat/models/website_visitor.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, api, fields, _
+from odoo.exceptions import UserError
+import json
+
+
+class WebsiteVisitor(models.Model):
+    _inherit = 'website.visitor'
+
+    livechat_request_ids = fields.One2many('im_livechat.request', 'visitor_id', string='Pending Chat Request', readonly=True)
+    livechat_operator_id = fields.Many2one('res.partner', compute='_compute_livechat_operator_id', store=True,
+                                           string='Speaking with', search='_search_livechat_operator_id')
+    mail_channel_ids = fields.One2many('mail.channel', 'livechat_visitor_id',
+                                       string="Visitor's livechat channels", readonly=True)
+    session_count = fields.Integer('# Sessions', compute="_compute_session_count")
+
+    @api.depends('mail_channel_ids.livechat_active', 'mail_channel_ids.livechat_operator_id')
+    def _compute_livechat_operator_id(self):
+        sql = """SELECT livechat_visitor_id, livechat_operator_id, livechat_active
+                    FROM mail_channel
+                    WHERE livechat_visitor_id in %s
+                    AND livechat_active = true"""
+        self.env.cr.execute(sql, [tuple(self.ids)])
+        results = self.env.cr.dictfetchall()
+        visitor_operator_map = {int(result['livechat_visitor_id']): int(result['livechat_operator_id']) for result in results}
+        for visitor in self:
+            visitor.livechat_operator_id = visitor_operator_map[visitor.id] if visitor.id in visitor_operator_map else False
+
+    @api.depends('mail_channel_ids')
+    def _compute_session_count(self):
+        sessions = self.env['mail.channel'].read_group([('livechat_visitor_id', 'in', self.ids)], ['livechat_visitor_id'], ['livechat_visitor_id'])
+        sessions_count = {session['livechat_visitor_id'][0]: session['livechat_visitor_id_count'] for session in sessions}
+        for visitor in self:
+            visitor.session_count = sessions_count[visitor.id] if visitor.id in sessions_count else 0
+
+    def action_send_chat_request(self):
+        """ Send a chat request to website_visitor(s).
+        This creates a chat_request and a mail_channel with livechat active flag.
+        But for the visitor to get the chat request, the operator still has to speak to the visitor.
+        The visitor will receive the chat request the next time he navigates to a website page.
+        (see _handle_visitor_response for next step)"""
+        # check if visitor is available
+        unavailable_visitors_count = self.env['mail.channel'].search_count([('livechat_visitor_id', 'in', self.ids), ('livechat_active', '=', True)])
+        if unavailable_visitors_count:
+            if len(self) == 1:
+                raise UserError(_('This user is not available. Please refresh the page to get latest visitor status.'))
```

Comment:
I would vote for a more generic "Recipients are not available. Please refresh ..." and avoid singular / plural dictinction